.github/workflows/nightly.yml

```diff
@@ -0,0 +1,62 @@
+name: Nightly
+
+on:
+  workflow_dispatch: # to be able to run it manually
+  schedule:
+    - cron: '0 0 * * *' # once a day
+
+jobs:
+  evaluateChanges:
+    name: Evaluate changes for run or skipping nightly build
+    runs-on: ubuntu-latest
+    outputs:
+      SHOULD_BUILD: ${{ steps.check.outputs.shouldBuild }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Check
+        id: check
+        run: |
+          if [[ $(git rev-list --after="24 hours" main) ]]; then
+            echo shouldBuild=true >> $GITHUB_OUTPUT
+          else
+            echo shouldBuild=false >> $GITHUB_OUTPUT
+          fi
+
+
+  build:
+    name: Generate Nightly Build
+    runs-on: ubuntu-latest
+    needs: evaluateChanges
+    if: needs.evaluateChanges.outputs.SHOULD_BUILD == 'true'
+    env:
+      TELEGRAM_CHAT_ID: ${{ secrets.TELEGRAM_CHAT_ID }}
+      TELEGRAM_TOKEN: ${{ secrets.TELEGRAM_TOKEN }}
+      SCHEDULED_BUILD_VARIANT: ${{ secrets.SCHEDULED_BUILD_VARIANT }}
```

Comment:
`TELEGRAM_CHAT_ID` and `SCHEDULED_BUILD_VARIANT` can be repository variables instead of secrets?
```yaml
      TELEGRAM_CHAT_ID: ${{ vars.TELEGRAM_CHAT_ID }}
      TELEGRAM_TOKEN: ${{ secrets.TELEGRAM_TOKEN }}
      SCHEDULED_BUILD_VARIANT: ${{ vars.SCHEDULED_BUILD_VARIANT }}
```